packages/webapp/pages/join.tsx

```diff
@@ -0,0 +1,235 @@
+import Logo from '@dailydotdev/shared/src/components/Logo';
+import { ProfilePicture } from '@dailydotdev/shared/src/components/ProfilePicture';
+import {
+  Button,
+  ButtonSize,
+} from '@dailydotdev/shared/src/components/buttons/Button';
+import {
+  FlexCentered,
+  FlexCol,
+  FlexRow,
+} from '@dailydotdev/shared/src/components/utilities';
+import { GET_REFERRING_USER_QUERY } from '@dailydotdev/shared/src/graphql/users';
+import { ReferralCampaignKey } from '@dailydotdev/shared/src/hooks';
+import { graphqlUrl } from '@dailydotdev/shared/src/lib/config';
+import {
+  downloadBrowserExtension,
+  isDevelopment,
+} from '@dailydotdev/shared/src/lib/constants';
+import { PublicProfile } from '@dailydotdev/shared/src/lib/user';
+import request from 'graphql-request';
+import React, { ReactElement, useContext, useEffect, useState } from 'react';
+import BrowsersIcon from '@dailydotdev/shared/icons/browsers.svg';
+import { Image } from '@dailydotdev/shared/src/components/image/Image';
+import { cloudinary } from '@dailydotdev/shared/src/lib/image';
+import PlayIcon from '@dailydotdev/shared/src/components/icons/Play';
+import { IconSize } from '@dailydotdev/shared/src/components/Icon';
+import CloseButton from '@dailydotdev/shared/src/components/CloseButton';
+import {
+  ModalKind,
+  ModalSize,
+} from '@dailydotdev/shared/src/components/modals/common/types';
+import { Modal } from '@dailydotdev/shared/src/components/modals/common/Modal';
+import { useRouter } from 'next/router';
+import AuthContext from '@dailydotdev/shared/src/contexts/AuthContext';
+import useMedia from '@dailydotdev/shared/src/hooks/useMedia';
+import { laptop } from '@dailydotdev/shared/src/styles/media';
+import { GetServerSideProps } from 'next';
+import { NextSeo, NextSeoProps } from 'next-seo';
+import { defaultOpenGraph } from '../next-seo';
+
+type PageProps = {
+  referringUser: Pick<PublicProfile, 'id' | 'name' | 'image'>;
+  campaign: ReferralCampaignKey;
+};
+
+const Page = ({ referringUser, campaign }: PageProps): ReactElement => {
+  const router = useRouter();
+  const { user } = useContext(AuthContext);
+  const [isVideoOpen, setVideoOpen] = useState(false);
+  const isDesktop = useMedia([laptop.replace('@media ', '')], [true], false);
+
+  const onVideoModalClose = () => {
+    setVideoOpen(false);
+  };
+
+  useEffect(() => {
+    if (!user?.id) {
+      return;
+    }
+
+    router.replace('/');
+  }, [user?.id, router]);
+
+  useEffect(() => {
+    document.cookie = [
+      `join_referral=${referringUser.id}:${campaign}`,
+      `max-age=${1 * 365 * 24 * 60 * 60}`,
+      isDevelopment ? undefined : 'Secure',
+      `domain=${process.env.NEXT_PUBLIC_DOMAIN}`,
```

Comment:
Will it be `app.daily.dev` or `daily.dev`?
We need another property called `SameSite` and set it to `Lax` so it will be accessible from all subdomains

Reply:
From what I can see it is `daily.dev`, confirmed it is on preview.

It is accessible for subdomains already since `Domain=.daily.dev`

I'all add Lax to to match the sharing of cookies for other requests since I see we do use it.

Reply:
Oh maybe I'm wrong but if I recall correctly when Lax was not presented the extension couldn't reach it.
An interesting experiment for later (out of this scope) is to check whether the Lax is really needed for the extension for example when it comes to our authentication cookie